### PR TITLE
Implemented the 'safe' changes derived from fuzzing

### PR DESCRIPTION
**Pull request by 0xfede7c8**

### CODE REVIEW

File: dbutil.h [hunk 1]
```diff
@@ -91,4 +91,11 @@ char * expand_homedir_path(const char *inpath);
 
 void fsync_parent_dir(const char* fn);
 
+#if DROPBEAR_MSAN
+/* FD_ZERO seems to leave some memory uninitialized. clear it to avoid false positives */
```
Comment (dgutson) [inline dbutil.h]: So it seems this is just to silence the tool...
In this case sorry and please revert this change.

File: common-kex.c [hunk 2]
```diff
@@ -687,6 +687,9 @@ void kexecdh_comb_key(struct kex_ecdh_param *param, buffer *pub_them,
 	/* K, the shared secret */
 	buf_putmpint(ses.kexhashbuf, ses.dh_K);
 
+	ecc_free(Q_them);
```
Comment (dgutson) [inline common-kex.c:690]: This seems the only important change.

**Review by dgutson — APPROVED** [review]

File: dbutil.h (hunk 1 above)
Comment (0xfede7c8) [reply dbutil.h]: Reverting.
